### PR TITLE
JDK level compilation switched to v7 for Eclipse m2e

### PR DESCRIPTION
**Pull request by axel3rd**
Since commit ea5e38c17986be0786389111a4b8d9b99350c0a2, JDK v7 is used
for compilation.
Some class (Files, Path, ...) are now used and require JDK v7.
This change is mainly to link a JDK v7 when Eclipse m2e is used (v6 used
before, so compilation problem, manuel update required).

### CODE REVIEW

**Comment by michael-o** [comment]
The PR is incomplete to me. Look at 6397c3e439528cd1c319c7e3eae448768f7a4552, two new classes have been introduced. They can be easily deprecated or removed. They also maybe other spots.

**Comment by rfscholte** [comment]
labelled as bug? I would say enhancement...

**Comment by michael-o** [comment]
I don't thing so because enforcer requires 1.7. Why should MCOMPILER require 1.6 only? This looks like a mismatch or bug. Unless I do not fully understand the issue.

**Comment by axel3rd** [comment]
> two new classes have been introduced

Good catch, thanks.
* I will delete [Java7Detector](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/Java7Detector) (if v7 considered as minimal level version)
* [Java7FileUtil](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/Java7FileUtil) has been already deleted previously
* Perhaps [NioFiles](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/NioFiles.java) should be reviewed

> Unless I do not fully understand the issue.

Basically this feature was to have the correct JDK version linked in Eclipse when project is imported :-)

**Comment by axel3rd** [comment]
[DirectoryScanner.isSymbolicLink](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/DirectoryScanner.java#L732) seems bugged: only parent tested, not the complete file => should be 
1. fixed or deleted (`NioFiles.isSymbolicLink` could be now used natively by end users) ?
1. and :
   1. In a dedicated PR
   1. In a separated commit of PR
   1. In the globale commit of PR
?

**Comment by michael-o** [comment]
@rfscholte  that's the bug:

    $ mvn help:effective-pom | grep maven.compiler.
        <maven.compiler.source>1.6</maven.compiler.source>
        <maven.compiler.target>1.6</maven.compiler.target>

**Comment by michael-o** [comment]
@axel3rd Incomplete tests should be separate PR. `NioFiles` shall be deprecated because they are public.

**Comment by rfscholte** [comment]
> I don't thing so because enforcer requires 1.7. Why should MCOMPILER require 1.6 only?

This is intended. By using source+target all classes get the right byte-code version and can be used by JDK.  However, by compiling with JDK 7 we can classes from this distribution. Normally you would call it "leaking" and the animal-sniffer-maven-plugin prevents you from doing so, but now we are using this feature. Java 9 comes with a better option: [JEP 247: Compile for Older Platform Versions](http://openjdk.java.net/jeps/247)

**Comment by axel3rd** [comment]
> Incomplete tests should be separate PR. 

Ok

> NioFiles shall be deprecated because they are public.

After reviewing the content, perhaps could stay unchanged. This is an utility class who "simplify" the usage of `Files` and `Path` (with usage if `File`).

**Comment by michael-o** [comment]
@rfscholte Makes sense now, though ugly. Will be leave at 6 now or move to 7? If not, we can close this PR.

**Comment by rfscholte** [comment]
Let's call it forward compatibility :) I'm not sure if this is the right moment to require Java 7, Plexus Utils is used a lot.

**Comment by michael-o** [comment]
@axel3rd I think this PR has to be dropped because it won't do what we want. I think the best is a profile for m2e action to override the `javaVersion`.

**Comment by axel3rd** [comment]
PR updated with java 1.7 compilation level restricted to m2e Eclipse plugin usage (title updated).
For , [DirectoryScanner.isSymbolicLink](https://github.com/codehaus-plexus/plexus-utils/blob/master/src/main/java/org/codehaus/plexus/util/DirectoryScanner.java#L732) bug, see #28.

**Comment by michael-o** [comment]
@rfscholte Any objections to merge this? Looks good to me now.

**Comment by rfscholte** [comment]
I see that Eclipse is trying to pick the best matching JDK. You'll only see this issue when you have JDK6 installed. However, if you run any goal from the IDE, its bytecode version will be wrong.
This profile is only there to fool Eclipse. Eclipse should respect the requireJavaVersion enforcer-rule instead.
So I'm -0. I don't think it's worth it (number of developers which also have JDK6 is to low). But to prevent the compilation error of NioFiles in Eclipse, I would prefer proper Javadoc in that class. The solution would be to manually change the JRE System Library.

**Comment by michael-o** [comment]
@rfscholte Agreed. Though, Eclipse ignores the Enforcer rule here. @axel3rd Do you have JDK 6 installed?

**Comment by axel3rd** [comment]
@michael-o : Yes, I have the JDK 6 configured in my Eclipse. I have updated the profile comment with the exact cause. Even if this is a m2e bug, this workaround can't be added ? It is boring to configure the *JRE System Library* to *JavaSE-1.7* manually under Eclipse on any checkout ; But I can continue to do that if this profile is not sustainable ...

**Comment by axel3rd** [comment]
Thanks :-)
